lib/install/management/appliance.go

```diff
@@ -93,6 +77,35 @@ func (d *Dispatcher) removeApplianceIfForced(conf *metadata.VirtualContainerHost
 	return nil
 }
 
+func (d *Dispatcher) deleteVM(vm *vm.VirtualMachine) error {
+	var err error
+	// ignore power off error cause VM can be powered off already
+	if _, err = tasks.WaitForResult(d.ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
+		return vm.PowerOff(ctx)
+	}); err != nil {
+		log.Debugf("Failed to power off existing appliance for %s, try to remove anyway", err)
+	}
+	_, err = tasks.WaitForResult(d.ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
+		return vm.Destroy(ctx)
+	})
+	if err != nil {
+		err = errors.Errorf("Failed to destroy vm %s: %s", vm.Reference(), err)
+		return err
+	}
+	return nil
+}
+
+func (d *Dispatcher) deleteDatastoreFiles(conf *metadata.VirtualContainerHostConfigSpec) error {
```

Comment:
We should log the full `[ds] path` of the directory we're attempting to delete so that the viadmin knows where to do manual cleanup.